custom/icds_reports/sqldata/exports/growth.py

```diff
@@ -0,0 +1,179 @@
+    current_month_stunting_column, \
+    current_month_wasting_column, phone_number_function, india_now
+from dateutil.relativedelta import relativedelta
+
+
+class GrowthExport(ExportableMixin, IcdsSqlData):
+    title = 'Child Growth Tracking'
+    table_name = 'child_health_monthly_view'
+
+    def get_excel_data(self, location):
+
+        def _check_case_presence(case_id, column, data_dict):
+            return data_dict[case_id][column] if case_id in data_dict.keys() else "N/A"
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
+            "valid_in_month": 1
+        }
+        initial_month = self.config['month']
+
+        if self.loc_level == 5:
+            filters['awc_id'] = location
+            order_by = ('awc_name',)
+        elif self.loc_level == 4:
+            filters['supervisor_id'] = location
+            order_by = ('awc_name',)
+        elif self.loc_level == 3:
+            filters['block_id'] = location
+            order_by = ('supervisor_name', 'awc_name')
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name')
+        elif self.loc_level == 1:
+            filters['state_id'] = location
+            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name')
+        else:
+            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
+
+        query_set_1 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+        data_initial_month = query_set_1.values('state_name', 'district_name', 'block_name', 'supervisor_name',
+                                                'awc_name', 'awc_site_code', 'person_name', 'dob', 'mother_name',
+                                                'mother_phone_number', 'pse_days_attended', 'lunch_count',
+                                                'current_month_nutrition_status',
+                                                current_month_wasting_column(self.beta),
+                                                current_month_stunting_column(self.beta), 'case_id')
+
+        filters["month"] = initial_month - relativedelta(months=1)
+        query_set_2 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+        data_past_month = query_set_2.values('case_id', 'pse_days_attended', 'lunch_count',
+                                             'current_month_nutrition_status',
+                                             current_month_wasting_column(self.beta),
+                                             current_month_stunting_column(self.beta), 'case_id')
+        data_past_month = {item['case_id']: item for item in data_past_month}
+
+        filters["month"] = initial_month - relativedelta(months=2)
+        query_set_3 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
```

Comment:
For weighing the performance. Could you put the cost of following here:
1. If you pull all the three months data together
2. Doing it separately like this.

Reply:
Added!